test/rr-tests.js

```diff
@@ -0,0 +1,286 @@
+      }
+        
+    }
+  };
+
+  const eventDataS3 = {
+    object: {
+      apiVersion: 'deploy.razee.io/v1alpha2',
+      kind: 'RemoteResource',
+      metadata: {
+        name: 'rr',
+        namespace: 'default'
+      },
+      spec: {
+        clusterAuth: {
+          impersonateUser: 'razeedeploy'
+        },
+        backendService: 's3',
+        iam: {
+          url: 'https://iam.cloud.ibm.com/identity/token',
+          grantType: 'urn:ibm:params:oauth:grant-type:apikey',
+          apiKey: 'testApiKey'
+        },
+        requests: [
+          {
+            options: {
+              url: 'https://s3.us.cloud-object-storage.appdomain.cloud/bucket/test-config.yaml'
+            }
+          }
+        ]
+      }
+        
+    }
+  };
+
+  const eventDataS3_1 = {
+    object: {
+      apiVersion: 'deploy.razee.io/v1alpha2',
+      kind: 'RemoteResource',
+      metadata: {
+        name: 'rr',
+        namespace: 'default'
+      },
+      spec: {
+        clusterAuth: {
+          impersonateUser: 'razeedeploy'
+        },
+        backendService: 's3',
+        iam: {
+          url: 'https://iam.cloud.ibm.com/identity/token',
+          grantType: 'urn:ibm:params:oauth:grant-type:apikey',
+          apiKey: 'testApiKey'
+        },
+        requests: [
+          {
+            options: {
+              url: 'https://s3.us.cloud-object-storage.appdomain.cloud/bucket/'
+            }
+          }
+        ]
+      }
+        
+    }
+  };
+
+  it('Create generic RemoteResourceController', async function () {
+    // backendServiceFactory should create RemoteResourceController for generic backendService
+    const controller = setupController(eventData);
+    assert(controller instanceof RemoteResourceController);
+  });
+
+  it('RRGitController single file request', async function() {
+    // RRGitController added() should correctly assemble request option
+    const controller = setupController(eventDataGit);
+    assert(controller instanceof RemoteResourceGitController);
+    await controller.added();
+    const requests = controller.data.object.spec.requests;
+
+    assert.equal(requests[0].options.url, 'https://raw.githubusercontent.com/razee-io/RemoteResource/master/test/test-configs/test-config.yaml');
+    assert.deepEqual(requests[0].options.headers, { 'User-Agent': 'razee-io', Accept: 'application/octet-stream' });
+    assert(requests[0].splitRequestId);
+  });
+
+  it('RRGitController multiple files request', async function() {
+    // RRGitController added() should correctly assemble request options for multiple files
+    const controller = setupController(eventDataGit1);
+    assert(controller instanceof RemoteResourceGitController);
+    await controller.added();
+    const requests = controller.data.object.spec.requests;
+
+    assert.equal(requests.length, 3);
+    assert.equal(requests[0].options.url, 'https://raw.githubusercontent.com/razee-io/RemoteResource/master/test/test-configs/test-config.yaml');
```

Comment:
Once you have the request mocked, and the mock data available, set this up to loop and check each array element instead of manually indexing